k9mail/src/main/java/com/fsck/k9/ui/crypto/MessageCryptoHelper.java

```diff
@@ -344,18 +340,29 @@ public void onProgress(int current, int max) {
         });
     }
 
+    /**
+     * Common operations preceding {@link #onCryptoOperationReturned(MimeBodyPart)}
+     * @param result
+     * @param decryptedPart
+     */
+    private void onCryptoOpReturnedPreops(Intent result, MimeBodyPart decryptedPart) {
+        cancelableBackgroundOperation = null;
+        currentCryptoResult = result;
+        onCryptoOperationReturned(decryptedPart);
+    }
+
     private void callAsyncDetachedVerify(Intent intent) throws IOException, MessagingException {
         OpenPgpDataSource dataSource = getDataSourceForSignedData(currentCryptoPart.part);
 
         byte[] signatureData = MessageDecryptVerifier.getSignatureData(currentCryptoPart.part);
+        if (signatureData != null)
+            intent.putExtra(OpenPgpApi.EXTRA_DETACHED_SIGNATURE, signatureData);
         intent.putExtra(OpenPgpApi.EXTRA_DETACHED_SIGNATURE, signatureData);
```

Comment:
This line and the previous make no sense. Also, did you check what actually happens when this parameter is null?